### PR TITLE
Only query latest ISCN version by default

### PR DESCRIPTION
**Pull request by nnkken**
This PR needs parallel migration.
Procedure:
1. run poller with new image, keep http service on old image
2. poller will create new table (`iscn_latest_version`) and maintain the table when inserting new ISCN
3. run `tx-indexer migrate setup-iscn-version-table` to setup `iscn_latest_version` table for old ISCN records (took ~11 minutes on my machine with production data)
4. update http service to run on new image

Also note that the queries (either for ISCN records or NFT related queries joining ISCN records) are now by default returning only records according to latest ISCN version, unless `all_iscn_versions=true` is specified.
This makes some weird situations, e.g.
`/iscn/records?iscn_id=iscn://likecoin-chain/zmpGY5Z8A1tn4vqSJJktmyvuLFv0xDcKFFfHKHmN4hc/2` returns nothing since the latest version is > 2.
`/iscn/records?iscn_id=iscn://likecoin-chain/zmpGY5Z8A1tn4vqSJJktmyvuLFv0xDcKFFfHKHmN4hc/2&all_iscn_versions=true` is needed for this record.
This is weird since the query is specifying the ISCN ID, but since there are many different conditions under the same endpoint, currently it is done like this...

### CODE REVIEW

**Comment by nnkken** [comment]
I think requiring all current ISCN ID queries to add `&all_iscn_versions=true` is too harsh... Let me improve that tomorrow.

**Comment by nnkken** [comment]
Previous issue fixed, ready for review.

File: db/iscn_test.go [hunk 1]
```diff
@@ -244,7 +245,7 @@ func TestIscnPagination(t *testing.T) {
 		}: 1,
 	}
 	for p, expectedNextKey := range table {
-		res, err := QueryIscnList(conn, p)
+		res, err := QueryIscnList(conn, p, true)
```
Comment (williamchong) [inline db/iscn_test.go:248]: No test case for `false` case?

**Comment by nnkken** [comment]
Some test cases updated to cover more cases on latest ISCN only queries.
But current test cases are too messy (since they are extracted from production data), so it's hard to setup test cases for new features.
May need another PR for refactoring test cases (e.g. localized test data for each test)

**Review by williamchong — APPROVED** [review]

**Pull request merged**